plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -272,7 +277,30 @@ private TupleDomain<HiveColumnHandle> buildTupleDomainColumnHandle(EntryType ent
                         type)),
                 ColumnType.REGULAR,
                 column.getComment());
-        return TupleDomain.withColumnDomains(ImmutableMap.of(handle, Domain.notNull(handle.getType())));
+
+        ImmutableMap.Builder<HiveColumnHandle, Domain> domains = ImmutableMap.<HiveColumnHandle, Domain>builder()
+                .put(handle, Domain.notNull(handle.getType()));
+        if (entryType == ADD) {
+            partitionConstraint.getDomains().ifPresent(domain -> domain.forEach((key, value) -> domains.put(toPartitionValuesParsedField(handle, key), value)));
```

Comment:
what if `partitionvalues_parsed` is not present in the checkpoint?
the Parquet reader will return no rows, right?

I think the condition on `partitionvalues_parsed.<part-column>` should be generated only when `partitionvalues_parsed` is present in the file footer.

Reply:
Parquet reader ignores the condition if the field doesn't exist in my understanding. 

https://github.com/trinodb/trino/blob/a24475e3011c22cc107db0878d6b308cededb823/plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java#L455-L459